os/arch/arm/src/amebasmart/amebasmart_flash.c

```diff
@@ -284,6 +285,7 @@ static ssize_t amebasmart_write(FAR struct mtd_dev_s *dev, off_t offset, size_t
 {
 	size_t addr;
 	ssize_t result;
+	DEBUGASSERT(nbytes > 0);

```

Comment:
Do we need ASSERT in this case? return 0 is not enough?

Reply:
Sorry I missed out that datatype, seems the only case to be considered here is 0.
If we are just checking for 0, then I think return warning message should be enough. In my opinion, it should be tolerable as we can just do nothing and return from there

Reply:
So, final your opinion is?
I will follow yours because it's a DEBUGASSERT.

Reply:
Okay, let's go with DEBUGASSERT. Since return error and adding error logs will also induce more code size.